src/App.js

```diff
@@ -31,7 +36,9 @@ type AppState = {
     programVer: number,
     settings: AppSettings,
     dashConnectionStatus: DeviceConnectionStatus,
-    speechRecognitionOn: boolean
+    mode: string,
```

Comment:
Is this the state value that is controlling which editor is visible? Let's name it something more descriptive like `editorMode`. Let's also move it inside `settings` and add a new type to constrain the allowed values:

```
type EditorMode = 'block' | 'text';

type AppSettings = {
    dashSupport: boolean,
    editorMode: EditorMode,
    language: string
}
```